lib/python/qmk/cli/doctor/check.py

```diff
@@ -44,7 +44,12 @@ def _check_arm_gcc_version():
     version_number = ESSENTIAL_BINARIES['arm-none-eabi-gcc']['output'].strip()
     cli.log.info('Found arm-none-eabi-gcc version %s', version_number)
 
-    return CheckStatus.OK  # Right now all known arm versions are ok
+    parsed_version = _parse_gcc_version(version_number)
+    if (parsed_version['major'], parsed_version['minor']) < (10, 3):
+        cli.log.warning('{fg_yellow}We do not recommend arm-none-eabi-gcc older than 10.3.x. Upgrading to 10.3.x or higher is recommended.')
+        return CheckStatus.WARNING
+
+    return CheckStatus.OK  # arm versions less than 10.3.x cause issues on sn32
```

Comment:
What is the reason for this?

Reply:
the reason for this is occasional and random hardfaults when built against those versions in arm m0. The toolchain version that first fixed it was `arm-none-eabi-gcc (GNU Arm Embedded Toolchain 10-2020-q4-major)`. Bumping the version one up makes sure the right one is selected ( up until q2 it was buggy). No known issues exist with current latest version, tested extensively with every major release from 10.3 onwards.

while my original notes are lost on this, I believe this to be the [bug](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=91816)